### PR TITLE
Dev: first commit

### PR DESCRIPTION
**Pull request by sum414ever**

### CODE REVIEW

File: src/main/java/edu/curssor/moviesPlatform/controller/MovieController.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @GetMapping("movie/stats/{category}")
+    public ResponseEntity<List<Movie>> getMovieByCategory(@PathVariable String category){
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .body(movieService.getMovieByCategory(category));
+    }
+
+    @GetMapping("movie/stats/rating/highest")
```
Comment (oleskafeska) [inline src/main/java/edu/curssor/moviesPlatform/controller/MovieController.java]: I propose to combine those 2 methods to one and put ordering as an input parameter.
In movie service create a method which decides according to parameter in which way data should be sorted.

But it's not required, you could use this one as well.

File: src/main/java/edu/curssor/moviesPlatform/entities/Movie.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
+    private String id;
+
+    private String name;
+
+    private String category;
+
+    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
+    private Category categoryDB;
```
Comment (oleskafeska) [inline src/main/java/edu/curssor/moviesPlatform/entities/Movie.java]: rename this to:
enum - category
string - categoryLabel

File: src/main/java/edu/curssor/moviesPlatform/enums/Category.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    ADV("adventure"),
+    COM("comedy"),
+    CRM("crime"),
+    DRM("drama"),
+    HOR("horror"),
+    SCF("science fiction"),
+    WES("western"),
+    OTH("other");
+
+    private String category;
```
Comment (oleskafeska) [inline src/main/java/edu/curssor/moviesPlatform/enums/Category.java]: rename string  to 'label'

File: src/main/java/edu/curssor/moviesPlatform/enums/Category.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    Category(String category) {
+        this.category = category;
+    }
+
+    public static Category getCategoryFromString(String category) {
```
Comment (oleskafeska) [inline src/main/java/edu/curssor/moviesPlatform/enums/Category.java:20]: getCategoryFromString -> getLabel

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,3 @@
+server.port=8080
```
Comment (oleskafeska) [inline src/main/resources/application.properties]: 8080 it's already default port :)

Comment (sum414ever) [reply src/main/resources/application.properties]: Fixed

File: src/main/java/edu/curssor/moviesPlatform/enums/Category.java (hunk 4 above)
Comment (sum414ever) [reply src/main/java/edu/curssor/moviesPlatform/enums/Category.java:20]: Fixed

File: src/main/java/edu/curssor/moviesPlatform/enums/Category.java (hunk 3 above)
Comment (sum414ever) [reply src/main/java/edu/curssor/moviesPlatform/enums/Category.java]: Fixed

File: src/main/java/edu/curssor/moviesPlatform/entities/Movie.java (hunk 2 above)
Comment (sum414ever) [reply src/main/java/edu/curssor/moviesPlatform/entities/Movie.java]: Fixed

File: src/main/java/edu/curssor/moviesPlatform/controller/MovieController.java (hunk 1 above)
Comment (sum414ever) [reply src/main/java/edu/curssor/moviesPlatform/controller/MovieController.java]: Fixed

File: src/main/java/edu/curssor/moviesPlatform/enums/Category.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        for (Category cat : Category.values()) {
+            if (cat.getCategory().equalsIgnoreCase(category))
+                return cat;
+        }
+        return Category.OTH;
+    }
+
+    public static String getLabel(Category category){
```
Comment (omilovskiy) [inline src/main/java/edu/curssor/moviesPlatform/enums/Category.java:28]: 1. Do not write the logic in the enums
2. Why u just can't get label by this way: `category.getLabel();`? 😄

File: src/main/java/edu/curssor/moviesPlatform/service/implementation/MovieServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    public Movie findMovieById(String id) {
+        return movieRepository.findById(id).orElseThrow(() -> new DataNotFoundException("Movie with id " + id + " not found"));
+    }
+
+    @Override
+    public Movie updateMovieById(String id, Movie movieDTO) {
+
+        Movie editedMovie = findMovieById(id);
+
+        String newName = Objects.nonNull(movieDTO.getName())
```
Comment (omilovskiy) [inline src/main/java/edu/curssor/moviesPlatform/service/implementation/MovieServiceImpl.java:64]: move ternary operator into a separate method

File: src/main/java/edu/curssor/moviesPlatform/enums/Category.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+        for (Category cat  : Category.values()) {
+            if (cat.getCategory().equalsIgnoreCase(category.getCategory()))
```
Comment (omilovskiy) [inline src/main/java/edu/curssor/moviesPlatform/enums/Category.java]: btw, always wrap body into scopes by this way:

```
if (cat.getCategory().equalsIgnoreCase(category.getCategory())) {
    return cat.getCategory();
}
```

File: src/main/java/edu/curssor/moviesPlatform/service/implementation/MovieServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    @Override
+    public List<Movie> getMovieByCategory(String category) {
+        Category cat = Category.getCategoryFromString(category);
+        return movieRepository.findAllByCategory(cat);
+    }
+
+    @Override
+    public List<Movie> getAllMovieByRating(int sortedParameter) {
```
Comment (omilovskiy) [inline src/main/java/edu/curssor/moviesPlatform/service/implementation/MovieServiceImpl.java:107]:
```
public List<Movie> getAllMovieByRating(int sortedParameter) {
        List<Movie> movies = movieRepository.findAll();

        List<Movie> sortedMovies = movies.stream()
            .sorted(Comparator.comparing(movie -> movie.getRate().getAvgRate()))
            .collect(Collectors.toList());

        return sortedParameter == SOME_CONSTANT 
            ? Collections.reverse(sortedMovies) 
            : sortedMovies;
}
```

File: src/main/java/edu/curssor/moviesPlatform/enums/Category.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    OTH("other");
+
+    private String label;
+
+    Category(String category) {
+        this.label = category;
+    }
+
+    public static Category getCategoryFromString(String category) {
+        for (Category cat : Category.values()) {
```
Comment (omilovskiy) [inline src/main/java/edu/curssor/moviesPlatform/enums/Category.java:21]: What about:
`Category catery = Category.valueOf(string);` ?

**Review by omilovskiy — APPROVED** [review]
Approved 😉 
But take a look on my comments.

File: src/main/java/edu/curssor/moviesPlatform/enums/Category.java (hunk 8 above)
Comment (sum414ever) [reply src/main/java/edu/curssor/moviesPlatform/enums/Category.java]: Fixed

File: src/main/java/edu/curssor/moviesPlatform/service/implementation/MovieServiceImpl.java (hunk 7 above)
Comment (sum414ever) [reply src/main/java/edu/curssor/moviesPlatform/service/implementation/MovieServiceImpl.java:64]: Fixed

**Pull request merged**